spec/components/header/navigation_component_spec.rb

```diff
@@ -28,4 +49,46 @@
   specify "extra resources are last" do
     expect(component.all_resources.map(&:title)).to end_with(extra_resources.values)
   end
+
+  context "when using a desktop browser" do
+    it "renders a dropdown menu for category links" do
+      expect(page).to have_css("#secondary-navigation > div.category-links > ol.category-links-list > li[id='page-five-category-1-desktop'] > button")
```

Comment:
is there a less brittle and more readable way to test these things? eg

```
within('#secondary-navigation') do
  within('div.category-links') do
    expect(page).to have_button('whatever the button text is')
  end
end
```

Reply:
can't seem to make this approach work? - the within blocks are skipped over